Sources/TokamakGTK/Shapes/Shape.swift

```diff
@@ -0,0 +1,158 @@
+// Copyright 2020 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//  Created by Morten Bek Ditlevsen on 29/12/2020.
+//
+
+import CGTK
+import CGDK
+import TokamakCore
+
+func createPath(from elements: [Path.Element], in cr: OpaquePointer) {
+  var current: CGPoint = .zero
+  var start: CGPoint = .zero
+  for element in elements {
+    switch element {
+    case let .move(to: p):
+      cairo_move_to(cr, p.x, p.y)
+      current = p
+      start = p
+
+    case let .line(to: p):
+      cairo_line_to(cr, p.x, p.y)
+      current = p
+
+    case .closeSubpath:
+      cairo_close_path(cr)
+      current = start
+
+    case let .curve(to: p, control1: c1, control2: c2):
+      cairo_curve_to(cr, c1.x, c1.y, c2.x, c2.y, p.x, p.y)
+      current = p
+
+    case let .quadCurve(to: p, control: c):
+      let c1 = CGPoint(x: (current.x + 2 * c.x) / 3, y: (current.y + 2 * c.y) / 3)
+      let c2 = CGPoint(x: (p.x + 2 * c.x) / 3, y: (p.y + 2 * c.y) / 3)
+      cairo_curve_to(cr, c1.x, c1.y, c2.x, c2.y, p.x, p.y)
+      current = p
+    }
+  }
+}
+
+extension _ShapeView: ViewDeferredToRenderer {
+  public var deferredBody: AnyView {
+    return AnyView(WidgetView(build: { _ in
+      let w = gtk_drawing_area_new()
+      bindAction(to: w!)
+      return w!
+    }) {
+    })
+  }
+
+  func bindAction(to drawingArea: UnsafeMutablePointer<GtkWidget>) {
+    drawingArea.connect(signal: "draw", closure: { widget, cr in
+      cairo_save(cr)
+
+      let width = gtk_widget_get_allocated_width(widget)
+      let height = gtk_widget_get_allocated_height(widget)
+
+      let c = (style as? Color) ?? foregroundColor ?? Color.black
+
+      // XXX TODO, don't know why my environment doesn't work.
+      var environment = EnvironmentValues()
+      environment[_ColorSchemeKey] = .light
+
+      let rgba = _ColorProxy(c).resolve(in: environment)
+      var color = GdkRGBA(red: Double(rgba.red),
```

Comment:
Could this be made an extension on `Color` so we can easily get a `GdkRGBA` value from a resolved `Color` in other places too?

Reply:
Ah, yes - of course! Thanks!

Reply:
Done.